helpers/DeliveryHelper.php

```diff
@@ -33,6 +33,7 @@
 use oat\taoProctoring\model\monitorCache\implementation\DeliveryMonitoringService;
 use oat\taoQtiTest\models\event\QtiTestChangeEvent;
 use qtism\runtime\tests\AssessmentTestSessionState;
+use oat\taoAct\model\runner\HeartbeatService;
```

Comment:
Please do not directly use the heartbeat here, as it is a customer specific. You should rather use an abstraction to handle the last activity. Thus an implementation of this abstraction can be done to handle the heartbeat, and this implementation should be registered from a customer extension.